Ylijokic/npm mysql

ylijokic opened the pull request:
Updated the remaining routes for the user, asset, and folder tables. 

You'll see 4 different types of routes in each JS file of the routes directory:
router.get routes for Selecting data
router.post for Inserting data
router.put for Updating data
router.delete for Deleting data

I don't really know how to test the routes without having some front end UI. I might just create some pages using Handlebar template engine just because that's what I've used in the past.

hooskers commented on routes/asset.js:
```diff
@@ -70,25 +55,69 @@ router.get('/:id', (req, res) => {
-// });
+// POST request to Create an asset
+router.post('/', (req, res) => {
+  let mysql = req.app.get('mysql');
```
You should try defaulting to `const` instead of `let` for variable declarations. You should really only use `let` if you are going to reassign a different value to the variable. The idea is that always using `const` will prevent you from accidentally reassigning a variable.

hooskers commented on routes/asset.js:
```diff
@@ -70,25 +55,69 @@ router.get('/:id', (req, res) => {
+    req.body.fileSize,
+    req.body.fileType
+  ];
+  sql = mysql.pool.query(sql, inserts, (err, result) => {
```
You should avoid reassigning variables unless you really need to. And, even then, you shouldn't reassign a value of a different type.
In this case, since you're not using the `sql` variable anywhere below here, you don't actually need to assign the output of `mysql.pool.query` to a variable.
You can just run it like a standalone function and when it's done it calls the callback function you provided.

hooskers commented on routes/asset.js:
```diff
@@ -70,25 +55,69 @@ router.get('/:id', (req, res) => {
+  sql = mysql.pool.query(sql, inserts, (err, results) => {
+    if (err) {
+      // console.log(error)
+      res.write(JSON.stringify(error));
```
I don't see `error` defined anywhere. I think maybe you meant to do:
```suggestion
      res.write(JSON.stringify(err));
```
instead?

hooskers commented on routes/folder.js:
```diff
@@ -67,25 +55,54 @@ router.get('/:id', (req, res) => {
+  let inserts = [req.body.name, req.body.creator, req.params.id];
+  sql = mysql.pool.query(sql, inserts, (err, results) => {
+    if (err) {
+      res.write(JSON.stringify(error));
```
```suggestion
      res.write(JSON.stringify(err));
```

hooskers commented on routes/folder.js:
```diff
@@ -67,25 +55,54 @@ router.get('/:id', (req, res) => {
+  sql = mysql.pool.query(sql, inserts, (err, results) => {
+    if (err) {
+      // console.log(error)
+      res.write(JSON.stringify(error));
```
```suggestion
      res.write(JSON.stringify(err));
```

hooskers commented on routes/user.js:
```diff
@@ -64,25 +55,54 @@ router.get('/:id', (req, res) => {
+  let inserts = [req.body.email, req.body.pw, req.params.id];
+  sql = mysql.pool.query(sql, inserts, (err, results) => {
+    if (err) {
+      res.write(JSON.stringify(error));
```
```suggestion
      res.write(JSON.stringify(err));
```

hooskers commented on routes/user.js:
```diff
@@ -64,25 +55,54 @@ router.get('/:id', (req, res) => {
+  sql = mysql.pool.query(sql, inserts, (err, results) => {
+    if (err) {
+      // console.log(error)
+      res.write(JSON.stringify(error));
```
```suggestion
      res.write(JSON.stringify(err));
```

ylijokic commented:
I changed all the let variable declarations to const. Also fixed up some of the syntax errors I had.  Should be ready to commit to master now I think.

hooskers commented:
Cool! Feel free to merge whenever you're ready
